### PR TITLE
sidecar scope matches ns for envoy filter and authn/z

### PR DESCRIPTION
**Pull request by stevenctl**
Continuation of https://github.com/istio/istio/pull/25403 for https://github.com/istio/istio/issues/25364

Adds the following to the list of "namespace scope" resources for a sidecar:
- EnvoyFilter
- AuthorizationPolicy
- ~PeerAuthentication~ will have issues
- RequestAuthentication

Still need to validate this scoping is accurate for these resource types.

### CODE REVIEW

**Comment by howardjohn** [comment]
@yangminzhu can you validate assumptions here? Basically, what this will do is a change in a Security API in namespace `foo` will only trigger a push to proxies in namespace `foo`. A change in root namespace will still trigger to all proxies.

**Comment by stevenctl** [comment]
Per the docs:

> The order of application of these EnvoyFilters is as follows: **all EnvoyFilters in the config root namespace, followed by all matching EnvoyFilters in the workload’s namespace.**

> Authorization Policy scope (target) is determined by “metadata/namespace” and an optional “selector”.

authn policy
> Apply the policy to the namespace of the workload it selects

File: pilot/pkg/model/sidecar.go [hunk 1]
```diff
@@ -41,7 +41,11 @@ var (
 	}
 
 	sidecarScopeNamespaceConfigTypes = map[resource.GroupVersionKind]struct{}{
-		gvk.Sidecar: {},
+		gvk.Sidecar:               {},
+		gvk.EnvoyFilter:           {},
+		gvk.AuthorizationPolicy:   {},
```
Comment (rshriram) [inline pilot/pkg/model/sidecar.go:46]: I didn't follow the previous PR but remember that EnvoyFilter is additive.
envoyFilter for Workload is (root namespace envoy filters, namespace level envoy filters, plus namespace level workload specific envoy filters (i.e. with matching labels)).

I am not sure if the request authentication is additive or not. You can trace the code to determine the behavior

Comment (stevenctl) [reply pilot/pkg/model/sidecar.go:46]: Yeah, that's pretty much the behavior except we'll trigger a push for workload specific filters even if they don't match.

Comment (howardjohn) [reply pilot/pkg/model/sidecar.go:46]: Yeah we can technically get even finer grained with the workload selector but I don't think its worth the effort right now - with this change we will support full namespace isolation which is 99% of what we need anyways, and the last 1% will be pretty complex

File: pilot/pkg/model/sidecar.go [hunk 2]
```diff
@@ -41,7 +41,11 @@ var (
...
+		gvk.PeerAuthentication:    {},
```
Comment (yangminzhu) [inline pilot/pkg/model/sidecar.go:47]: Looks good for RequestAuthentication and AuthorizationPolicy, but not sure if there is any implications for autoMTLS with PeerAuthentication, @incfly can you confirm. thanks.

Comment (incfly) [reply pilot/pkg/model/sidecar.go:47]: PeerAuthentication policy for server side only affect its own namespace all entire mesh if the CR is in root namespace. 

But if `foo` ns changed from STRICT to plaintext, we expect the client side's auto mTLS config in bar namespace to respect that as well, should receive CDS push for the update to remove the auto mTLS config.

Will this PR break above requirement? @howardjohn

**Pull request merged**